### ISSUE TITLE
Endurance-Page-Cache and PHP Edge Plugin

### ISSUE THREAD

**Issue by BLUX88** [bluehost/endurance-page-cache#34]
Hi all,

I have a Wordpress site that occasionally encounters issues with the navigation bar distorting and the site completely breaking where any thing clicked on simply downloads a PHP file. I've come to realize it is due to an "endurance page cache" and "PHP edge plugin". Neither of these are listed in the Plugins directory.

I originally built this site on a staging site hosted by Bluehost and then transferred it to the client's domain which is hosted by Dreamhost. I'm wondering if Bluehost installed these two plugins without my permission on my staging site and also how to get them uninstalled without breaking the site. 

My final question is if I should install WP-Super-Cache and if that will cause the site any issues? 

Thank you all for the help!!

**Comment by MikeHansenMe** [comment]
Hey @BLUX88 chances are that the issue is the php-edge plugin. This is a plugin built for the Bluehost platform to ensure you are running php 7 & in the future maybe a newer version. This plugin will not work on a Dreamhost account. The Endurance Cache plugin is also built specifically for the nginx cache layers at Bluehost. These plugins are not on the repo for that reason (they only work here at BH). They are in the mu-plugins folder because we manage them for our customers (updates etc). I order to remove them you can delete the files in the wp-contents/mu-plugins/ folder. 

Installing Super-Cache will work as expected and does not conflict with the EPC plugin but unless you are on a Bluehost server, you are not getting much benefit from the EPC plugin.

**Comment by iseabail** [comment]
I moved a site from BlueHost to Dreamhost and was just beating my head against the wall.  This solved my problem!  THANK YOU! THANK YOU! THANK YOU!

**Comment by wpscholar** [comment]
@finsdad We realize the frustration that this can cause. We are working to have a reliable method of checking if we are on a Bluehost environment so that we don't run any of these customizations outside of a Bluehost server even if the code is copied over.

**Comment by john-f-chamberlain** [comment]
@wpscholar in reality this type of plugin shouldn't ever be used to begin with. What if PHP 7.4 comes out tomorrow and is installed on BlueHost, but absolutely destroys half the WordPress installations / plugins that are available?

I'd recommend leaving WordPress on whatever version of PHP their account is running at the time WordPress is installed and not upgrading PHP automatically. Obviously customers could still switch to a newer version via cPanel on their own.

This type of practice causes unneeded confusion and has a very real potential to break someone's site without any easy way for a layman to fix it.

**Comment by MikeHansenMe** [comment]
php-edge is no longer installed on new installs. That was a short-term solution to make sure our customers new WP sites were getting the newest version of PHP until we could get the server defaults increased. We now have the defaults increased and no longer have a need for php-edge. Sites that have been on bluehost for awhile may still have it in place.  

To your question about when new versions come out and us updating the plugin. That is no different than if the default of the server changed. However, when this plugin was introduced it was about 5.x.x -> 7.x.x and only on new installs. We have no intention to use this long term but rather keep our servers running versions that are up to date.

We do understand the frustration this can cause, especially if you move the site. As I mentioned before this was a temporary solution. That is no longer installed by default.

**Comment by DustinAllan** [comment]
I just did an new wordpress install and ran into this issue. It is still definitely installed by default. I wanted to be on php 7.2 and this edge plug-in would keep forcing me back to 7.0.33. Even the bluehost support wasn't able to figure this out. After four calls into support, and 4 times being forced back to version 7...I figured it out myself. Frustrating is an understatement. Especially when four separate support specialists dont know anything about it.

**Comment by DustinAllan** [comment]
I just did an new wordpress install and ran into this issue. It is still definitely installed by default. I wanted to be on php 7.2 and this edge plug-in would keep forcing me back to 7.0.33. Even the bluehost support wasn't able to figure this out. After four calls into support, and 4 times being forced back to version 7...I figured it out myself. Frustrating is an understatement. Especially when four separate support specialists dont know anything about it.
